integration-tests/tests/src/tests/migrations.ts

```diff
@@ -0,0 +1,521 @@
+          newRealm.deleteModel("Test");
+        },
+      });
+
+      expect(realm.schema.length).equals(1); // model should remain
+      expect(realm.objects("Test").length).equals(0); // objects should be gone
+
+      realm.close();
+
+      realm = new Realm({ schema: [TestSchema], schemaVersion: 2 });
+      expect(realm.objects("Test").length).equals(0);
+
+      realm.close();
+    });
+    it("should ignore deleteModel with non-existing models", () => {
+      realm = new Realm({
+        schema: [TestSchema],
+        schemaVersion: 1,
+        onMigration: function (oldRealm, newRealm) {
+          newRealm.deleteModel("NonExistingModel");
+        },
+      });
+
+      realm.close();
+
+      realm = new Realm({ schema: [TestSchema], schemaVersion: 2 });
+      expect(realm.objects("Test").length).equals(1);
+
+      realm.close();
+    });
+  });
+
+  describe("Delete model with relationships", () => {
+    beforeEach(() => {
+      Realm.clearTestState();
+    });
+
+    const PersonWithDogsSchema = {
+      ...PersonSchema,
+      properties: {
+        ...PersonSchema.properties,
+        dogs: {
+          type: "linkingObjects",
+          objectType: "Dog",
+          property: "owner",
+        },
+      },
+    };
+
+    it("should work correctly", () => {
+      let realm = new Realm({ schema: [PersonWithDogsSchema, DogSchema] });
+
+      realm.write(function () {
+        const owner = realm.create<IPerson>("Person", { name: "John Doe", age: 10 });
+        realm.create<IDog>("Dog", {
+          name: "My Dog",
+          owner: owner,
+          age: 5,
+        });
+      });
+
+      expect(realm.objects("Person").length).equals(1);
+      expect(realm.objects("Dog").length).equals(1);
+      expect(realm.objects<IDog>("Dog")[0].owner.name).equals("John Doe");
+      expect(realm.objects<IPerson>("Person")[0].dogs[0].name).equals("My Dog");
+
+      realm.close();
+
+      realm = new Realm({
+        schema: [DogSchema, PersonSchema],
+        schemaVersion: 1,
+        onMigration: function (oldRealm, newRealm) {
+          expect(() => {
+            // Deleting a model which is target of linkingObjects results in an exception
+            newRealm.deleteModel("Person");
+          }).throws("Table is target of cross-table link columns");
+        },
+      });
+
+      expect(realm.objects("Person").length).equals(1);
+      expect(realm.objects("Dog").length).equals(1);
+
+      realm.close();
+
+      realm = new Realm({
+        schema: [DogSchema, PersonSchema],
+        schemaVersion: 2,
+        onMigration: function (oldRealm, newRealm) {
+          // deleting a model which isn't target of linkingObjects works fine
+          newRealm.deleteModel("Dog");
+        },
+      });
+
+      expect(realm.objects("Person").length).equals(1);
+      expect(realm.objects("Dog").length).equals(0);
+      expect(realm.objects<IPerson>("Person")[0].dogs).equals(undefined);
```

Comment:
This used to check equivalent of `expect(realm.objects<IPerson>("Person")[0].dogs.length).equals(0);` but seems like this behavior has changed now so I am now testing for this (which in my opinion makes more sense in a way). Is this intentional?

Reply:
That doesn't seem correct, dogs is not supposed to be undefined and should always default to an empty list.  The old behavior was correct.

Reply:
I was able to find my error here and correct it, it had to do with the schemas I was passing.